In [1]:
filePath = "arnetminer/AP_train.txt"
#filePath = "sample-data"
file = open(filePath,'r')

distinctAuthors=[]
distinctVenues=[]
distinctPublications=[]
indexes=[] #indexes are unique and !=""

for line in file:
    
    # capture index
    if line.startswith("#index"):
        index = line[6:].strip()
        indexes.append(index)        
            
    # capture publication name        
    if line.startswith("#*"):
        publication = line[2:].strip()
        if publication != "":
            distinctPublications.append(publication)
            
    # capture list of authors        
    if line.startswith("#@"):
        authors = line[2:].strip().split(";")
        for author in authors:
            author = author.strip()
            if author != "":
                distinctAuthors.append(author)
    
    # capture venues
    if line.startswith("#c"):
        venue = line[2:].strip()
        if venue != "":
            distinctVenues.append(venue)
                            
file.close()

print("No. of distinct authors: ", len(set(distinctAuthors)))
print("No. of distinct Venues: ", len(set(distinctVenues)))
print("No. of distinct Publications: ", len(indexes))
print("No. of distinct Publication names: ", len(set(distinctPublications)))

No. of distinct authors:  1478733
No. of distinct Venues:  255685
No. of distinct Publications:  1976815
No. of distinct Publication names:  1929486


As per my understanding of the dataset, these values seem incorrect. 

The distinct venue count could be incorrect because, the same venues are repeated with different names. For example, the venue associated with the conference "Principles and Practice of Knowledge Discovery in Databases" have been repeated multiple times under different names.

No. of distinct Publications are 1929486. Whereas, No. of distinct indexes are 1976815. This proves that some of the Publlication names are missing and are equal to "".

Similarily, the distinct authors count could be incorrect because, the same author have been reported under different names. For example, the author "Jorge Jesus Gomez-Sanz" have been reported as "Jorge J. Gomez-Sanz", "Jorge Gomez-Sanz". Also, another author "Henryk Jan Komorowski" have been reported under "H. Jan Komorowski", "Jan Komorowski", "Henryk J. Komorowski".